pulp_container/tests/functional/api/test_push_signatures.py

```diff
@@ -0,0 +1,119 @@
+"""Tests that verify that an image signature can be pushed to Pulp."""
+import gnupg
+import base64
+import json
+import requests
+import pytest
+import tempfile
+
+from pulp_container.tests.functional.constants import REGISTRY_V2_REPO_PULP
+from pulp_container.constants import SIGNATURE_TYPE
+
+KEY = "https://raw.githubusercontent.com/pulp/pulp-fixtures/master/common/GPG-PRIVATE-KEY-pulp-qe"
+
+
+@pytest.fixture(scope="session")
+def tempdir():
+    """A temporary directory."""
+    with tempfile.TemporaryDirectory() as directory:
+        yield directory
+
+
+@pytest.fixture(scope="session")
+def gpg_metadata(tempdir):
+    """A fixture for a GPG instance and related metadata."""
+    private_key = requests.get(KEY).text
+
+    gpg = gnupg.GPG(gnupghome=tempdir)
+
+    gpg.import_keys(private_key)
+
+    fingerprint = gpg.list_keys()[0]["fingerprint"]
+    keyid = gpg.list_keys()[0]["keyid"]
+
+    gpg.trust_keys(fingerprint, "TRUST_ULTIMATE")
+
+    return gpg, fingerprint, keyid
+
+
+@pytest.fixture
+def distribution(
+    registry_client,
+    local_registry,
+    container_distribution_api,
+    gpg_metadata,
+    add_to_cleanup,
+):
+    """Return a distribution created after pushing a signed content to the Pulp Registry."""
+    registry_client.pull(f"{REGISTRY_V2_REPO_PULP}:manifest_a")
+
+    image_path = f"{REGISTRY_V2_REPO_PULP}:manifest_a"
+    gpg, fingerprint, keyid = gpg_metadata
+
+    with registry_client._client.machine.env(GNUPGHOME=gpg.gnupghome):
+        local_registry.push(image_path, "test-1:manifest_a", "--sign-by", keyid)
+
+        # push the same image for the second time with a different signature (timestamp)
+        local_registry.push(image_path, "test-1:manifest_a", "--sign-by", keyid)
```

Comment:
This looks nice.
BTW, heads up: On the other PR it was suggested to call this `tag_and_push`. Don't change it yet.

Reply:
I'm now wondering why this works without the pulp-smash change.